### PR TITLE
SYNCOPE-872 - type extensions on user form

### PR DESCRIPTION
**Pull request by mteo**

### CODE REVIEW

File: client/enduser/src/main/java/org/apache/syncope/client/enduser/resources/UserSelfCreateResource.java [hunk 1]
```diff
@@ -13,16 +13,26 @@
  * software distributed under the License is distributed on an
  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  * KIND, either express or implied.  See the License for the
+ * LOG.info("Update ... AAAAAAAAAAAAAAAAAAAAA {}", json);
```
Comment (ilgrosso) [inline client/enduser/src/main/java/org/apache/syncope/client/enduser/resources/UserSelfCreateResource.java]: Please remove such (debug?) statements

File: client/enduser/src/main/java/org/apache/syncope/client/enduser/resources/UserSelfUpdateResource.java [hunk 2]
```diff
@@ -57,8 +63,8 @@ protected ResourceResponse newResourceResponse(final IResource.Attributes attrib
                 throw new IllegalArgumentException("Entered captcha is not matching");
             }
 
-            UserTO userTO = MAPPER.readValue(request.getReader().readLine(), UserTO.class);
-            LOG.debug("User {} id self-updating", userTO.getUsername());
+            String json = request.getReader().readLine();
```
Comment (ilgrosso) [inline client/enduser/src/main/java/org/apache/syncope/client/enduser/resources/UserSelfUpdateResource.java]: Don't see any reason for this change - and debug statement also went away: please revert this change.

File: client/enduser/src/main/resources/META-INF/resources/app/js/directives/dynamicPlainAttribute.js [hunk 3]
```diff
@@ -85,10 +83,10 @@ angular.module('self')
 
                   case "Date":
                     var dateInMs = $scope.user.plainAttrs[schema.key].values[index];
-                    console.debug("received date in milliseconds", dateInMs);
+//                    console.debug("received date in milliseconds", dateInMs);
```
Comment (ilgrosso) [inline client/enduser/src/main/resources/META-INF/resources/app/js/directives/dynamicPlainAttribute.js]: Uncomment or remove this debug statement

File: client/enduser/src/main/resources/META-INF/resources/app/js/directives/dynamicPlainAttribute.js [hunk 4]
```diff
@@ -85,10 +83,10 @@ angular.module('self')
...
                     if (dateInMs) {
                       var temporaryDate = new Date(dateInMs * 1);
-                      console.debug("date-time", temporaryDate);
+//                      console.debug("date-time", temporaryDate);
```
Comment (ilgrosso) [inline client/enduser/src/main/resources/META-INF/resources/app/js/directives/dynamicPlainAttribute.js]: Uncomment or remove this debug statement

**Comment by ilgrosso** [comment]
I have made some comments: please address them. then rebase (there are unresolved conflicts ATM) and finally sqash into a single commit.

Please also change the subject to something mentioning SYNCOPE-872 (to get automatically linked on JIRA),

**Comment by mteo** [comment]
manually committed with 343b3d400fd2457e2458c52eacd03af17c51dbac
